src/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+
+  configure_file(
+    buf.gen.yaml.in
+    buf.gen.yaml
+  )
+
+  add_custom_command(
+    OUTPUT
+    # Unfortunately, there isn't a good way to know in advance what
+    # files will be generated by invoking `buf generate`.  Instead, we
+    # just list the files here that we know we need in the
+    # `add_library` call below. But we will find in practice that more
+    # files are generated.
+    #
+    # TODO: We reiterate this list below in `add_library`. Can we DRY
+    # it sanely? We would needd to filter out (or add in) the headers.
+
+    # This list is needed for the core library.
+    ${PROTO_GEN_DIR}/gen/common/v1/common.grpc.pb.cc
+    ${PROTO_GEN_DIR}/gen/common/v1/common.grpc.pb.h
+    ${PROTO_GEN_DIR}/gen/common/v1/common.pb.cc
+    ${PROTO_GEN_DIR}/gen/common/v1/common.pb.h
+    ${PROTO_GEN_DIR}/gen/google/api/annotations.pb.cc
+    ${PROTO_GEN_DIR}/gen/google/api/annotations.pb.h
+    ${PROTO_GEN_DIR}/gen/google/api/http.pb.cc
+    ${PROTO_GEN_DIR}/gen/google/api/http.pb.h
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.grpc.pb.cc
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.grpc.pb.h
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.pb.cc
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.pb.h
+
+    # These are needed for `echo` example, which is awkward. These can
+    # probably be removed if that example is pulled.
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.grpc.pb.cc
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.grpc.pb.h
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.pb.cc
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.pb.h
+
+    COMMAND buf generate buf.build/viamrobotics/goutils --template buf.gen.yaml
+    COMMAND buf generate buf.build/googleapis/googleapis --template buf.gen.yaml --path google/rpc --path google/api
+    COMMAND buf generate buf.build/viamrobotics/api --template buf.gen.yaml --path common,component,robot,service,module
+    MAIN_DEPENDENCY buf.gen.yaml
+  )
+
+  add_custom_target(
+    update-static-protos
+    # This must be one of the files listed in `add_custom_command` above,
+    # but it doesn't matter which one, we just need to have a dependency edge
+    # into the files produced by that command.
+    #
+    # TODO: If we DRY as suggested above, this could just be the first list item.
+    #
+    DEPENDS gen/common/v1/common.grpc.pb.cc
+  )
+
+  add_custom_command(
+    TARGET update-static-protos
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${PROTO_GEN_DIR}/gen ${CMAKE_SOURCE_DIR}/src/gen
+  )
+endif(VIAMCPPSDK_USE_DYNAMIC_PROTOS)
+
+# I dislike exporting values via `CACHE` but for now we need this
+# state over in the `examples` directory. Hopefully when we do some
+# physical refactoring that can be eliminated.
+set(PROTO_GEN_DIR ${PROTO_GEN_DIR} CACHE INTERNAL "")
+
+
+# Declare the viamcpp library, and attach sources,
+# headers for installation
+add_library(viamcpp)
+
+target_sources(viamcpp
+  PRIVATE
+    ${PROTO_GEN_DIR}/gen/common/v1/common.grpc.pb.cc
```

Comment:
Folding the generated sources in here is one approach. Another would be to make a dedicated library for just the generated sources. There's some appeal to that, as it would let us deliver the SDK as a pair of libraries: one low-level that builds/exposes the protos, and another that wraps them up in a presentation layer.

I'd also like to take a closer look at how the `buf generate` tool works. Perhaps there is a way to get it to emit a manifest of what files were generated?

Reply:
I don't think it'll print a manifest of output, but you could probably do some wildcard expansion in cmake.

Alternately, buf DOES let you restrict generation to only specific paths. So if you don't need the entire API here, you could generate just the paths that are needed, and then only have to wildcard match those paths (and not worry about stray files.)

Reply:
I shied away from wildcard/globbing because we are definitely currently generating more than we need to (e.g. things used in the `echo` example). I like the path restriction idea assuming it can be made tight enough to produce only what we need. I've added a TODO.

Reply:
Resolving this discussion around narrowing the collection of `buf` generated files, as I don't think we need a better answer, yet.